vmm/src/config.rs

```diff
@@ -113,14 +114,17 @@ impl OptionParser {
         for option in options_list.iter() {
             let parts: Vec<&str> = option.split('=').collect();
 
-            if parts.len() != 2 {
-                return Err(OptionParserError::InvalidSyntax((*option).to_owned()));
-            }
-
             match self.options.get_mut(parts[0]) {
                 None => return Err(OptionParserError::UnknownOption(parts[0].to_owned())),
                 Some(value) => {
-                    value.value = Some(parts[1].trim().to_owned());
+                    if value.requires_value {
+                        if parts.len() != 2 {
+                            return Err(OptionParserError::InvalidSyntax((*option).to_owned()));
+                        }
+                        value.value = Some(parts[1].trim().to_owned());
+                    } else {
+                        value.value = Some(String::new());
```

Comment:
Why do we need to make it `Some()`? Seems like `None` would be a good fit for this case?

Reply:
It's the special case situation where we have "valueless" options, like "tty" or "off" for the console. They need to be present (so the value can't be `None`) but there is no value for them.

Reply:
Ah ok, otherwise they would be considered as not set I bet.